emitter/test/part1.test.ts

```diff
@@ -0,0 +1,121 @@
+import { describe, it, beforeEach, expect, vi } from 'vitest';
+import EventEmitter from '../Emitter';
+
+describe("Emitter", () => {
+    let emitter;
+    let tracker;
+
+    beforeEach(() => {
+        emitter = new EventEmitter();
+        tracker = vi.fn();
+    });
+
+    it("can trigger an event", () => {
+        const done = vi.fn();
+        emitter.on("Hello", done);
+        emitter.trigger("Hello");
+
+        expect(done).toHaveBeenCalled();
+    });
+
+    it("can trigger an event with data", () => {
+        const handler = vi.fn((data) => {
+            expect(data).toBe("World");
+        });
+
+        emitter.on("Hello", handler);
+        emitter.trigger("Hello", "World");
+
+        expect(handler).toHaveBeenCalled();
+    });
+
+    it("can trigger an event with no handlers", () => {
+        expect(() => emitter.trigger("Hello")).not.toThrow();
```

Comment:
Probably best to also check there were no unhandled rejections, you never know what code people write